logbook-jaxrs/src/test/java/org/zalando/logbook/jaxrs/ClientAndServerTest.java

```diff
@@ -0,0 +1,330 @@
+    Correlation<String,String> serverCorrelation = serverCorrelationCaptor.getValue();
+    clientRequest = clientCorrelation.getOriginalRequest();
+    clientResponse = clientCorrelation.getOriginalResponse();
+    serverRequest = serverCorrelation.getOriginalRequest();
+    serverResponse = serverCorrelation.getOriginalResponse();
+  }
+
+  @Override
+  protected Application configure() {
+    initMocks(this);
+    setupMocks();
+
+    return new ResourceConfig(TestWebService.class)
+        .register(
+            new ServerLoggingFilter(
+                Logbook.builder()
+                       // do not replace multi-part form bodies, which is the default
+                       .rawRequestFilter(replaceBody(BodyReplacers.stream()))
+                       // use our "writer" to capture the request with response objects on the server
+                       .writer(serverMockWriter)
+                       .build()
+            )
+        ).register(MultiPartFeature.class);
+  }
+
+  @BeforeEach
+  public void beforeEach() throws Exception {
+    super.setUp();
+
+    getClient().register(
+        new ClientLoggingFilter(
+            Logbook.builder()
+                   // do not replace multi-part form bodies, which is the default
+                   .rawRequestFilter(replaceBody(BodyReplacers.stream()))
+                   // use our "writer" to capture the request with response objects on the client
+                   .writer(clientMockWriter)
+                   .build()))
+               .register(MultiPartFeature.class);
+  }
+
+  @AfterEach
+  public void afterEach() throws Exception {
+    super.tearDown();
+  }
+
+  @Test
+  public void getWithPathAndQueryParamReturningTextPlain() throws Exception {
+    final String result = target("testws/testGet/first/textPlain")
+        .queryParam("param2", "second")
+        .request()
+        .get(String.class);
+
+    populateRequestsAndResponsesFromMockCaptors();
+
+    assertAll(
+        () -> assertEquals("param1=first param2=second", result),
+
+        // clientRequest
+        () -> assertEquals("GET", clientRequest.getMethod(), "clientRequest method"),
+        () -> assertEquals(LOCAL, clientRequest.getOrigin(), "clientRequest origin"),
+        () -> assertEquals("localhost", clientRequest.getRemote()),
+        () -> assertEquals("http", clientRequest.getScheme(), "clientRequest scheme"),
+        () -> assertEquals("localhost", clientRequest.getHost(), "clientRequest host"),
+        () -> assertEquals(Optional.of(this.getPort()), clientRequest.getPort()),
+        () -> assertEquals("/testws/testGet/first/textPlain", clientRequest.getPath(), "clientRequest path"),
+        () -> assertEquals("param2=second", clientRequest.getQuery(), "clientRequest query"),
+        () -> assertEquals("Jersey/2.27 (HttpUrlConnection 1.8.0_131)", clientRequest.getHeaders().get("User-Agent").get(0), "clientRequest userAgent"),
```

Comment:
You should remove version numbers here. It fails on travis right now, because it's running a different java version.